expression/expression.go

```diff
@@ -186,6 +186,17 @@ func IsEQCondFromIn(expr Expression) bool {
 	return len(cols) > 0
 }
 
+// HandleOverflowOnSelection handles Overflow errors when evaluating selection filters.
+// We should ignore overflow errors when evaluating selection conditions:
+//		INSERT INTO t VALUES ("999999999999999999");
+//		SELECT * FROM t WHERE v;
+func HandleOverflowOnSelection(sc *stmtctx.StatementContext, val int64, err error) (int64, error) {
+	if sc.InSelectStmt && err != nil && types.ErrOverflow.Equal(err) {
+		return -1, nil
```

Comment:
could you add a test case for this code branch?

Reply:
`TestIssue14159` will go to this branch.